x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/results/RestGetOverallBucketsAction.java

```diff
@@ -59,7 +59,10 @@ protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient
             if (restRequest.hasParam(Request.END.getPreferredName())) {
                 request.setEnd(restRequest.param(Request.END.getPreferredName()));
             }
-            request.setAllowNoJobs(restRequest.paramAsBoolean(Request.ALLOW_NO_JOBS.getPreferredName(), request.allowNoJobs()));
+            request.setAllowNoMatch(
+                restRequest.paramAsBoolean(
+                    Request.ALLOW_NO_MATCH.getPreferredName(),
+                    restRequest.paramAsBoolean(Request.ALLOW_NO_JOBS, request.allowNoMatch())));
```

Comment:
it seems to me that the default way to use these parameters via an API is through the request parameter (not in the body). 

Does this add a deprecation warning if the user provides `allow_no_jobs` as a URL param?